src/Bicep.Core/Semantics/Namespaces/AzNamespaceSymbol.cs

```diff
@@ -163,54 +208,122 @@ private static IEnumerable<FunctionOverload> GetAzOverloads(ResourceScopeType re
             // TODO: Add schema for return type
             yield return new FunctionOverloadBuilder("deployment")
                 .WithReturnType(GetDeploymentReturnType(resourceScope))
-                .WithFixedParameters()
+                .WithDescription("Returns information about the current deployment operation.")
                 .Build();
 
             yield return new FunctionOverloadBuilder("environment")
                 .WithReturnType(GetEnvironmentReturnType())
-                .WithFixedParameters()
+                .WithDescription("Returns information about the Azure environment used for deployment.")
                 .Build();
 
             // TODO: This is based on docs. Verify
+            // the resourceId function relies on leading optional parameters that are disambiguated at runtime
```

Comment:
>leading [](start = 49, length = 7)

Given that the leading arguments can only be disambiguated at runtime, this is the best approach I could think of. Open to better ideas, though.